app/Http/Middleware/RemoteUserAuthenticate.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Exception;
+use App\Models\User;
+use App\Models\LdapAd;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Contracts\Auth\Guard;
+
+/**
+ * Authenticate a user by using the server global variable.
+ *
+ * @author Wes Hulette <jwhulette@gmail.com>
+ *
+ * @since 5.0.0
+ * @see https://github.com/Adldap2/Adldap2-Laravel/blob/v4.0/src/Middleware/WindowsAuthenticate.php
+ */
+class RemoteUserAuthenticate
+{
+    /**
+     * The authenticator implementation.
+     *
+     * @var Guard
+     */
+    protected $auth;
+
+    /**
+     * The LDAP implementation.
+     *
+     * @var LdapAd
+     */
+    protected $ldap;
+
+    /**
+     * Returns the configured key to use for retrieving
+     * the username from the server global variable.
+     *
+     * @var string
+     */
+    const REMOTE_USER_KEY = 'REMOTE_USER';
+
+    /**
+     * Returns the configured key to use for retrieving
+     * the username from the server global variable.
+     *
+     * IIS Sometimes sends this
+     *
+     * @var string
+     */
+    const AUTH_USER_KEY = 'AUTH_USER';
+
+    /**
+     * Constructor.
+     *
+     * @param \Illuminate\Contracts\Auth\Guard $auth
+     * @param \App\Models\LdapAd               $ldap
+     */
+    public function __construct(Guard $auth, LdapAd $ldap)
+    {
+        $this->auth = $auth;
+        $this->ldap = $ldap;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @author Wes Hulette <jwhulette@gmail.com>
+     *
+     * @since 5.0.0
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure                 $next
+     *
+     * @return \Closure
+     */
+    public function handle($request, Closure $next)
+    {
+        if (!$this->auth->check() && $this->ldap->ldapSettings['login_remote_user_enabled']) {
```

Comment:
Why would remote user be part of LDAP? REMOTE_USER can be implemented in a few different ways, and they don't always directly relate to LDAP.

Reply:
It's not... I must have had a brain fart... LOL... I'll update the code and change the PR title.